vars.mk

```diff
@@ -13,11 +13,11 @@
 #   limitations under the License.
 
 # Pinned Versions
-GO_VERSION=1.16.3-alpine
-CLI_VERSION=20.10.2
-ALPINE_VERSION=3.12.2
-GOLANGCI_LINT_VERSION=v1.33.0-alpine
-GOTESTSUM_VERSION=0.6.0
+GO_VERSION=1.18-alpine
```

Comment:
@silvin-lubecki do you prefer to keep the go version pinned to minor releases? (`1.18.1` currently)? (same for the Dockerfile probably).

Slightly confused why we define defaults for them _both_ as a make var and as a default var in the Dockerfile though; I think if we'd change the Makefile to remove the `=$(VARIABLE)`, the build-args should take the value from the environment (_if set_), and otherwise take the default from the Dockerfile, i.e., something like:


```make
BUILD_ARGS:=--build-arg GO_VERSION \
    --build-arg CLI_VERSION \
    --build-arg ALPINE_VERSION \
    --build-arg GOLANGCI_LINT_VERSION) \
    --build-arg TAG_NAME \
    --build-arg GOTESTSUM_VERSION \
    --build-arg BINARY_NAME \
    --build-arg BINARY
```

Reply:
if I can jump in.. keep up with 1.18 (1.18.x) would be nice, since it uses the most recent version, and the minor changes from Go are usually security updates.

Regarding point 2, you're right, we were setting the dependencies at the Makefile and var.mk file. I updated the Dockerfile to only read the ARG values from the Makefile (that sets the ARG values). WDYT?

Reply:
I'd rather keep the defaults in the Dockerfile if that works, that way the Dockerfile itself has usable defaults, which allows it to be  built without having to pass specific build args.

Not near my computer to try if that works though ☺️

Reply:
Good catch @thaJeztah 👍